dask/dataframe/io/parquet/utils.py

```diff
@@ -500,12 +502,32 @@ def _normalize_index_columns(user_columns, data_columns, user_index, data_index)
     return column_names, index_names
 
 
-def _sort_and_analyze_paths(file_list, fs, root=False):
-    file_list = sorted(file_list, key=natural_sort_key)
+def _sort_and_analyze_paths(file_list, fs, root=False, key=natural_sort_key):
+    file_list = _maybe_sort_paths(file_list, key=key) if key else file_list
     base, fns = _analyze_paths(file_list, fs, root=root)
     return file_list, base, fns
 
 
+def _maybe_sort_paths(paths, key=natural_sort_key):
+    # Sort `paths` if key is not None
+    if key is None:
+        return list(paths)
+    elif callable(key):
+        return sorted(paths, key=key)
+    raise TypeError(f"Expected callable or None got {type(key)}")
+
+
+def _sort_glob_paths(paths, urlpath):
+    # Always use "natural" order to sort a list of
+    # paths that originated from a "glob" pattern.
+    # We assume this is the case if the `paths` list
+    # is longer than the user-provided `urlpath`.
+    urlpath = urlpath if isinstance(urlpath, list) else [urlpath]
+    if len(paths) > len(urlpath):
+        return _maybe_sort_paths(paths)
+    return paths
```

Comment:
@jrbourbeau - The plan to align with pandas feels pretty clean for the case of a user-provided path list.

One remaining complication is what to do about a glob pattern (e.g. `"/data/*.parquet"` or `["/data/month=10/*.parquet", "/data/month=9/*.parquet"]`). Since pandas does **not** support glob patterns, we cannot use `pd.read_parquet` as a guide for these cases.

In the latest commit ([7af2e61](https://github.com/dask/dask/pull/10061/commits/7af2e6197295aabe7993f3ac07d7a49960e6b935)), I am assuming that we should still use a natural sort on the "expanded" path list if there is a glob pattern anywhere in the user-provided path string or list. Do you think this makes sense? Or, should we start using "glob ordering" when users provide a glob pattern? (my intuition is that this would be a bit too much of a "breaking" change)

Reply:
Another note on the `["/data/month=10/*.parquet", "/data/month=9/*.parquet"]` case: It would be difficult to preserve the user-provided list order while using natural ordering to expand each of the glob patterns. This is because we lean on fsspec to expand the nested paths into a flat list.

Reply:
(though we can make it happen if a "hybrid" approach is clearly best - which **may** be the case)

Reply:
Apologies for the delayed response. 

> I am assuming that we should still use a natural sort on the "expanded" path list if there is a glob pattern anywhere in the user-provided path string or list

Yeah, I think I agree with that. Continuing to use natural ordering for a single glob string (e.g. `"/data/month=10/*.parquet"`) makes sense. My guess is we do that today so that files like

```
foo.parquet
├── part.0.parquet
├── part.1.parquet
...
└── part.9.parquet
``` 

have the number in the filepath correspond to the partition in the DataFrame. 

For the `["/data/month=10/*.parquet", "/data/month=9/*.parquet"]` case, what are your thoughts on just using our existing natural sorting in a loop so we maintain list order relative to each expanded glob string, but natural ordering within each glob string. For example, if we expand `["/data/month=10/*.parquet", "/data/month=9/*.parquet"]` we get something like:

```
[
"/data/month=10/00.parquet",
"/data/month=10/01.parquet",
"/data/month=10/02.parquet",
...,
"/data/month=9/00.parquet",
"/data/month=9/01.parquet",
"/data/month=9/02.parquet",
...
]
```

Reply:
>what are your thoughts on just using our existing natural sorting in a loop so we maintain list order relative to each expanded glob string, but natural ordering within each glob string.

Right. This was my initial thought as well, but I held off on implementing it after I realized we rely entirely on fsspec to expand paths, and would need to modify fsspec and/or re-implement custom utilities in dask.

Reply:
Nevermind. This shouldn't be too difficult.

Reply:
Okay - Latest commit should now "properly" handle the list-of-globs case discussed here (preserve list order while sorting expanded glob patterns for each item in the user-provided list).